rbi/documentation/versioned_annotations.md

```diff
@@ -0,0 +1,70 @@
+# Versioned RBI Annotations
+
+Many gems simultaneously maintain more than one version, often with different external APIs. If the gem's RBI
+annotation file does not match the version being used in your project, it can result in misleading type checking
+errors that slow down development.
+
+Starting in version 0.1.10, the [rbi gem](https://github.com/Shopify/rbi/) supports adding gem version comments
+to RBI annotations, allowing us to specify the gem versions that include specific methods and constants.
+[Tapioca](https://github.com/Shopify/tapioca) version 0.15.0 and later will strip out any parts of the annotation
+file that are not relevant to the current gem version when pulling annotations into a project.
+
+## Syntax
+
+To use this feature, add a comment in the following format directly above a method or constant:
+
+```ruby
+# @version > 0.2.0
+sig { void }
+def foo; end
+```
+
+The comment must start with a space, and then the string `@version`, followed by an [operator](#operators) and
+a version number. Version numbers must be compatible with Ruby's
+[`Gem::Version` specification](https://ruby-doc.org/current/stdlibs/rubygems/Gem/Version.html).
+
+Any method or constant that does not have a version annotation will be considered part of all versions.
+
+## Operators
+
+The following operators are accepted in version comments:
+
+| Symbol | Name  | Notes |
+---------------------------------
+| =      | Equal  | Only includes the specified gem version |
+| !=     | Not equal | Includes all gem versions except the one specified |
+| >      | Greater than | Includes all versions greater than the specified version |
+| >=     | Greater than or equal to | Includes the specified version and all greater versions |
+| <      | Less than | Includes all versions less than the specified version |
+| <=     | Less than or equal to | Includes the specified version and all lesser versions |
+| ~>     | [Pessimistic operator](https://thoughtbot.com/blog/rubys-pessimistic-operator) | Includes all versions between the specified version and the next version bump |
```

Comment:
While the statement here is correct, it doesn't cover all the cases, since there is a nuance for the single digit case:
> For the last example, single-digit versions are automatically extended with a zero to give a sensible result.

From: https://ruby-doc.org/3.3.4/stdlibs/rubygems/Gem/Version.html#class-Gem::Version-label-Preventing+Version+Catastrophe-3A